### PR TITLE
feat: deeplinks unit tests

### PR DESCRIPTION
**Pull request by estrattonbailey**
Screen Shot 2023-04-26 at 2 47 12 PM

### CODE REVIEW

File: src/analytics/__mocks__/index.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+export const analyticsV2 = {
+  identify: jest.fn(),
+  screen: jest.fn(),
+  track: jest.fn(),
+  setDeviceId: jest.fn(),
+  setCurrentWalletAddressHash: jest.fn(),
+  enable: jest.fn(),
+  disable: jest.fn(),
+  event,
+};
```
Comment (estrattonbailey) [inline src/analytics/__mocks__/index.ts:16]: Incomplete, but gets the job done for now

File: src/components/asset-list/RecyclerAssetList2/FastComponents/FastBalanceCoinRow.tsx [hunk 2]
```diff
@@ -110,7 +109,6 @@ const MemoizedBalanceCoinRow = React.memo(
           asset: item,
           fromDiscover: true,
           isFromWalletScreen: true,
-          longFormHeight: initialChartExpandedStateSheetHeight,
```
Comment (estrattonbailey) [inline src/components/asset-list/RecyclerAssetList2/FastComponents/FastBalanceCoinRow.tsx:113]: Removed all of these since they're `undefined` now

File: src/handlers/deeplinks.ts [hunk 3]
```diff
@@ -1,26 +1,25 @@
...
 import {
   checkIsValidAddressOrDomain,
   isENSAddressFormat,
 } from '@/helpers/validators';
 import { Navigation } from '@/navigation';
-import { scheduleActionOnAssetReceived } from '@/redux/data';
-import { emitAssetRequest, emitChartsRequest } from '@/redux/explorer';
 import Routes from '@/navigation/routesNames';
-import { ethereumUtils } from '@/utils';
+import ethereumUtils from '@/utils/ethereumUtils';
```
Comment (estrattonbailey) [inline src/handlers/deeplinks.ts:22]: Avoid the index imports, prevents importing a bunch of unrelated stuff, which could in turn require mocking

File: src/handlers/deeplinks.ts [hunk 4]
```diff
@@ -1,26 +1,25 @@
...
   walletConnectRemovePendingRedirect,
   walletConnectSetPendingRedirect,
 } from '@/redux/walletconnect';
+import { scheduleActionOnAssetReceived } from '@/redux/data';
+import { emitAssetRequest, emitChartsRequest } from '@/redux/explorer';
 import { fetchReverseRecordWithRetry } from '@/utils/profileUtils';
 import { defaultConfig } from '@/config/experimental';
 import { PROFILES } from '@/config/experimentalHooks';
-import { delay } from '@/helpers/utilities';
+import { delay } from '@/utils/delay';
```
Comment (estrattonbailey) [inline src/handlers/deeplinks.ts:15]: Putting this in its own file is similar: it doesn't import anything else, so no need to mock it most of the time.

File: src/handlers/deeplinks.ts [hunk 5]
```diff
@@ -142,6 +140,8 @@ export default async function handleDeeplink(
             url,
             query,
           });
+
+          break;
```
Comment (estrattonbailey) [inline src/handlers/deeplinks.ts:144]: Testing helped me notice that this was missing!

File: src/helpers/validators.ts
Comment (estrattonbailey) [inline src/helpers/validators.ts:1]: Just fixing things while I'm here

File: src/redux/__mocks__/store.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export default {
+  getState: jest.fn(),
+  dispatch: jest.fn(),
+};
```
Comment (estrattonbailey) [inline src/redux/__mocks__/store.ts:4]: Rudimentary for now

File: src/utils/__mocks__/delay.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+/**
+ * @desc Promise that will resolve after the ms interval
+ */
+export const delay = jest.fn().mockImplementation(
+  (ms: number): Promise<void> => {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+);
```
Comment (estrattonbailey) [inline src/utils/__mocks__/delay.ts:8]: Basically same function, but wrapped in a mock so we can spy on when it's called

File: src/utils/profileUtils.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 import colors from '../styles/colors';
-import { EthereumAddress } from '@/entities';
+import { EthereumAddress } from '@/entities/wallet';
```
Comment (estrattonbailey) [inline src/utils/profileUtils.ts:2]: Again, avoid index re-exports

**Review by brunobar79 — APPROVED** [review]
solid!

**Review by skylarbarrera — COMMENTED** [review]
👑 shit

**Review by skylarbarrera — APPROVED** [review]
